tools/format.py

```diff
@@ -65,7 +65,7 @@ def visit(path):
                                          style_config='google')
         return True
     else:
-        args = ['clang-format-6.0', '--style=file', '-i']
+        args = ['clang-format-15', '--style=file', '-i']
```

Comment:
I don't have anything against upgrading, but should we run a format sweep through all the repos and run it through CI for sanity if we're going to?

Note for others: If your dev desktop it still on Ubuntu <=20.04, you'll need to add LLVM repos to apt sources to get clang 15: https://stackoverflow.com/a/66654073

Reply:
Sure, that's a great idea. Re: 20.04, we can upgrade to clang-format-12 (latest version supported by both versions of Ubuntu) if we'd prefer to go that route for compatibility reasons. Figured this is mostly used internally so having the latest version is best, but that'd leave support for 20.04 users until the next Linux upgrade.